src/frontend/src/app/asset-readings/assets/assets.component.ts

```diff
@@ -68,11 +77,12 @@ export class AssetsComponent implements OnInit {
   }
 
   public setAssetCode(assetData) {
+    this.isChart = true;
+    this.isSummary = true;
     this.asset = assetData;
```

Comment:
assetData or this.assetData  and if its already already then why assign to another?

Reply:
it is required because assetData => object update on ngModelChange that need to be updated in class level variable.